### PR TITLE
Use TPC user configs (allowedTxPowers, userTxPowers)

### PR DESCRIPTION
**Pull request by wuweicaifb**
The TPC algorithms did not use some user config fields from DeviceConfig (the device configuration):
- allowedTxPowers
- userTxPowers

This PR uses those field in TPC algorithms.

Here's the list of major updates and some questions:
1. For available channel lists, we have different default channel lists for different channel widths and bands, but for TPC we don't. I use all the tx powers in the range of [`MIN_TX_POWER`, `MAX_TX_POWER`] as default for now. Any idea to have more specific default? @pohanhf 
2. `RandomTxPowerIntializer`: since different AP and band may have different available tx powers now, `setDifferentTxPowerPerAp` is removed. If we want to keep this arg, we can pick random one from the common tx power choices.
3. `LocationBasedOptimalTPC`: leverage the new method when computing tx powers, but there's a little functionally different. `userTxPowers` is not considered before, but is used to update `txPowerChoices` in the new method.
4. `MeasurementBasedApClientTPC` and `MeasurementBasedApApTPC`: these two compute the tx power directly without any choices, so the new method is not used in these two optimizers. We can check if the computed tx power is valid, i.e. in the `txPowerChoices`, but that's not done in this PR.

### CODE REVIEW

**Comment by wuweicaifb** [comment]
Update `DEFAULT_TX_POWER_CHOICES` as [20, 30] as was discussed with @pohanhf offline.

**Review by elludraon — COMMENTED** [review]
This looks okay to me, but maybe get a second review (@pohanhf?) for good measure.

File: src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java
Comment (wuweicaifb) [reply src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java:42]: As was mentioned in the point 2 of the summary, if we want to keep it, the easiest way is to randomly pick one from the common choices if it's true. If no common choices, raise an error. @pohanhf any idea?

Comment (pohanhf) [reply src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java:42]: Thanks for the question. I think we can keep it and use your idea to randomly pick one from the common choices. If no common choices, then just report some error messages to the users.

**Review by pohanhf — COMMENTED** [review]
It looks good to me. Thanks Wuwei for enabling this feature. Let's keep the setDifferentTxPowerPerAp arg so we have some flexibility for initialization.

File: src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java
Comment (jwoos) [inline src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java:135]: doesn't this violate the choices available since the set of tx power choices won't necessarily be the same across APs due to reading user config? e.g. AP A returns [1, 2, 3] and B returns [4, 5, 6] (just an example). Depending on iteration order we'd be violating either A or B's requested power.

Comment (wuweicaifb) [reply src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java:135]: Yes, you're right. When `setDifferentTxPowerPerAp` is false, `txPowerChoices` is computed as the shared tx powers across different APs, and then `dwfaultTxPower` is picked from that choices. If there's no shared tx powers (like you case), the default is used.

**Comment by wuweicaifb** [comment]
1. Leverage user configs in Measurement TPCs. Only pick TPC in the tx power choices. If the new TPC is not in the choice list, the nearest tx power is picked.

2. Change default tx power choices from [20, 30] to [0, 30]

File: src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java
Comment (jwoos) [inline src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java:272]: did you mean to leave this block?

**Review by jwoos — APPROVED** [review]
left a small comment about an unused block, looks good otherwise.

File: src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java
Comment (wuweicaifb) [reply src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java:272]: Yes. Just want `txPowerIndex` to be the first index where the element is larger than or equal to `newTxPower`. No further computation is needed in this block.

Comment (jwoos) [reply src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java:272]: makes sense, but I guess as a personal thing I would expect a for loop to be just giving iteration logic and then the check would come inside the block with a `break` or something

**Pull request merged**
